pkg/webhook/server.go

```diff
@@ -103,3 +107,27 @@ func (s *Server) setCert(name, namespace string) error {
 
 	return s.onCertChange(webhookCert)
 }
+
+func (s *Server) certCommonName() string {
+	return fmt.Sprintf("%s.%s.svc", s.name, s.namespace)
+}
+
+func (s *Server) handleCertChanges(ctx context.Context) {
+	log.Info().Str("webhook", s.name).Str("cn", s.certCommonName()).Msg("Listing for certificate rotations")
+	certRotationChan, unsub := s.cm.SubscribeRotations(s.certCommonName())
+	defer unsub()
+
+	for {
+		select {
+		case <-certRotationChan:
+			log.Debug().Str("webhook", s.name).Str("cn", s.certCommonName()).Msg("Certificate rotation was initiated for webhook")
+			if err := s.setCert(); err != nil {
+				// raise event?
```

Comment:
Could create an issue to discuss this and add a TODO comment

Reply:
Ya this is a broader problem we have throughout the code base.

Would you mind filing an issue along the lines of "Handle/record metric for errors better", and add a TODO here with the issue #

Reply:
https://github.com/openservicemesh/osm/issues/4956